index.js

```diff
@@ -20,75 +16,123 @@ let defaults = {
 }
 
 exports.register = function(server, options, next) {
-  let settings = _.clone(options);
+  // setup defaults:
+  var settings = _.clone(options);
   settings = _.defaults(settings, defaults);
-  let endpoint = settings.endpoint;
-  let log = settings.log;
-  let blocklist = settings.blocklist;
-  let validator = settings.validator;
+  var endpoint = settings.endpoint;
+  // you can pass your own logging function as options.log
+  // by default we use hapi's default logging and tag it to this plugin,
+  var log = settings.log ? settings.log : function log(message, tags){
+    var logTags = ['hapi-method-routes'];
+    // will push/combine any custom tags you want to add to the default tag:
+    if (_.isArray(tags)) logTags = _.union(logTags, tags);
+    if (_.isString(tags)) logTags.push(tags);
+    console.log("TAGS: %s MESSAGE: %s", logTags, message);
+    server.log(logTags, { message: message })
+  }
+  var whitelist = settings.whitelist;
+  var validator = settings.validator;
+  // list the methods we're exporting:
   log("methodsRoutePlugin is exporting the following methods at :" + endpoint);
   _.each(_.keys(server.methods), function(method){
-    log(method);
-  })
-  // will extract/convert the parameters and add the 'done' handler for us
-  function extractParamsFromRequest(request, done){
-   let params = _.map(request.params.params.split("/"), function(param){
-     return encodeURIComponent(param);
+    log(method, 'info');
+  });
+
+  // call this to extra params from request by GET or POST:
+  // note: I think we're mainly wanting to support JSON/POST requests
+  function extractParamsFromRequest(request){
+    console.log(request.method)
+    if (request.method.toLowerCase()=='post'){
+      log("using POST/JSON method ")
+      log(request.payload)
+      var params = _.map(request.payload.values, function(param){
```

Comment:
you are doing a map in both places, lets try to refactor the POST/GET params a bit